src/biokbase/narrative/tests/test_jobmanager.py

```diff
@@ -620,6 +620,97 @@ def test_cancel_job_lookup(self):
     #     self.assertTrue(self.jm._running_jobs[phony_id]['refresh'] == 0)
     #     self.assertIsNone(msg)
 
+    @mock.patch("biokbase.narrative.jobs.jobmanager.clients.get", get_mock_client)
+    def test_get_job_states(self):
+        job_ids = [
+            None,
+            None,
+            JOB_CREATED,
+            JOB_NOT_FOUND,
+            JOB_CREATED,
+            JOB_RUNNING,
+            JOB_TERMINATED,
+            JOB_COMPLETED,
+            BATCH_PARENT,
+            "",
+            JOB_NOT_FOUND,
+        ]
+
+        exp = {
+            **{
+                job_id: self.job_states[job_id]
+                for job_id in [JOB_CREATED, JOB_RUNNING, JOB_TERMINATED, JOB_COMPLETED, BATCH_PARENT]
+            },
+            JOB_NOT_FOUND: get_dne_job_state(JOB_NOT_FOUND)
+        }
+
+        res = self.jm.get_job_states(job_ids)
+        self.assertEqual(exp, res)
+
+    def test_get_job_states__empty(self):
+        with self.assertRaisesRegex(NoJobException, r"No job id\(s\) supplied"):
+            self.jm.get_job_states([])
+
+    def test_update_batch_job__dne(self):
+        with self.assertRaisesRegex(NoJobException, f"{JOB_NOT_FOUND} not registered"):
+            self.jm.update_batch_job(JOB_NOT_FOUND)
+
+    def test_update_batch_job__not_batch(self):
+        with self.assertRaisesRegex(NotBatchException, "Not a batch job"):
+            self.jm.update_batch_job(JOB_CREATED)
+
+        with self.assertRaisesRegex(NotBatchException, "Not a batch job"):
+            self.jm.update_batch_job(BATCH_TERMINATED)
+
+    @mock.patch("biokbase.narrative.jobs.jobmanager.clients.get", get_mock_client)
+    def test_update_batch_job__no_change(self):
+        job_ids = self.jm.update_batch_job(BATCH_PARENT)
+        self.assertEqual(BATCH_PARENT, job_ids[0])
+        self.assertCountEqual(BATCH_CHILDREN, job_ids[1:])
```

Comment:
Is it worth checking that the child IDs are the same?

Reply:
Not sure. I just have it implemented right now as returning them